src/lsstseries/analysis/structurefunction2.py

```diff
@@ -51,11 +51,12 @@ def calc_sf2(time, flux, err=None, band=None, lc_id=None, sf_method="basic", arg
     # `argument_container`. If any of these arguments are provided with
     # non-default values, we'll use those. Otherwise, we'll look inside
     # `argument_container` and use the values found there.
-    if band is None:
-        band = argument_container.band
+    band = _validate_band(band, flux, argument_container)
 
     if lc_id is None:
         lc_id = argument_container.lc_id
+    if lc_id is None:  # Still None after assignment?
```

Comment:
Use of the question makes this a little bit unclear. If I understand correctly, you are pointing to the person who is reading the code that this will be executed if there is no `lc_id` in the argument_container. I would say that more explicitly, `used if still None after assignment` or `used if not specified in after argument_container`

Reply:
Right, I see later you were more explicit in `_validate_band` function. I would apply same explanations (or same level of explanation) here.

Reply:
Yep, good comment. I committed some code last week so that I would have it handy, but it clearly wasn't polished. The most recent commit has corrected that, and brought that section of the input argument validation checks into sync with the `_validate_band` function.